crates/medea-coturn-telnet/src/connection.rs

```diff
@@ -0,0 +1,187 @@
+            CoturnCliResponse::Sessions(sessions) => Ok(sessions),
+            _ => Err(CoturnTelnetError::UnexpectedMessage(response)),
+        }
+    }
+
+    /// Closes session on Coturn server destroying this session allocations and
+    /// channels.
+    ///
+    /// 1. Sends [`CoturnCliRequest::CloseSession`] with specified session id.
+    /// 2. Awaits for [`CoturnCliResponse::Ready`].
+    pub async fn delete_session(
+        &mut self,
+        session_id: String,
+    ) -> Result<(), CoturnTelnetError> {
+        self.0
+            .send(CoturnCliRequest::CloseSession(session_id))
+            .await?;
+
+        // Await for `CoturnCliResponse::Ready`.
+        let response: CoturnCliResponse = self
+            .0
+            .next()
+            .await
+            .ok_or_else(|| CoturnTelnetError::Disconnected)??;
+        match response {
+            CoturnCliResponse::Ready => Ok(()),
+            _ => Err(CoturnTelnetError::UnexpectedMessage(response)),
+        }
+    }
+
+    /// Closes sessions on Coturn server destroying provided sessions
+    /// allocations and channels.
+    ///
+    /// For each provided session id:
+    /// 1. Sends [`CoturnCliRequest::CloseSession`] with specified session id.
+    /// 2. Awaits for [`CoturnCliResponse::Ready`].
+    pub async fn delete_sessions<T: IntoIterator<Item = String>>(
+        &mut self,
+        session_ids: T,
+    ) -> Result<(), CoturnTelnetError> {
+        for session_id in session_ids {
+            self.delete_session(session_id).await?;
+        }
+        Ok(())
+    }
+
+    /// Authenticates [`CoturnTelnetConnection`].
+    ///
+    /// 1. Awaits for [`CoturnCliResponse::EnterPassword`].
+    /// 2. Sends [`CoturnCliRequest::Auth`].
+    /// 3. Awaits for [`CoturnCliResponse::Ready`].
+    async fn auth(&mut self, pass: Bytes) -> Result<(), CoturnTelnetError> {
+        // Wait for `CoturnCliResponse::EnterPassword`;
+        let response = self
+            .0
+            .next()
+            .await
+            .ok_or_else(|| CoturnTelnetError::Disconnected)??;
+
+        if let CoturnCliResponse::EnterPassword = response {
+        } else {
+            return Err(CoturnTelnetError::UnexpectedMessage(response));
+        };
+
+        // Send `CoturnCliRequest::Auth` with provided password.
+        self.0.send(CoturnCliRequest::Auth(pass)).await?;
+
+        // Wait for `CoturnCliResponse::Ready`.
+        let response = self
+            .0
+            .next()
+            .await
+            .ok_or_else(|| CoturnTelnetError::Disconnected)??;
+        match response {
+            CoturnCliResponse::EnterPassword => {
+                Err(CoturnTelnetError::WrongPassword)
+            }
+            CoturnCliResponse::Ready => Ok(()),
+            _ => Err(CoturnTelnetError::UnexpectedMessage(response)),
+        }
+    }
+
+    /// Pings Coturn telnet server.
+    pub async fn ping(&mut self) -> Result<(), CoturnTelnetError> {
```

Comment:
Consider to provide `# Errors` section.

It's only a matter of time when the lint will complain on `async fn`s too.